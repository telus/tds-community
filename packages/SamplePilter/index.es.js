import './dist/index.css'
import SamplePilter from './dist/index.es'

export default SamplePilter
