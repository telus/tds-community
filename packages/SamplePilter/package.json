{
  "name": "@tds/community-sample-pilter",
  "version": "0.1.0",
  "description": "A branded pilter showcasing our practices",
  "license": "MIT",
  "private": true,
  "dependencies": {
    "@tds/core-box": "^1.0.1",
    "@tds/core-colours": "^1.0.0",
    "@tds/core-text": "^1.0.0",
    "prop-types": "^15.6.1"
  },
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  }
}
