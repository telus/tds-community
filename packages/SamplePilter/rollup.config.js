import configure from '../../config/rollup.config'
import { dependencies } from './package.json'

export default configure({
  input: './SamplePilter.jsx',
  dependencies,
})
