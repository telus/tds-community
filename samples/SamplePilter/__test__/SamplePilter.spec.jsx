import React from 'react'
import { render, shallow } from 'enzyme'

import SamplePilter from '../SamplePilter'

describe('SamplePilter', () => {
  const doShallow = props => shallow(<SamplePilter {...props}>Android</SamplePilter>)

  it('renders', () => {
    const pilter = render(<SamplePilter>Some Content</SamplePilter>)

    expect(pilter).toMatchSnapshot()
  })

  it('can have custom a11y text', () => {
    const pilter = doShallow({ a11yText: 'Sort by' })

    expect(pilter).toHaveProp('aria-label', 'Sort by Android')
  })

  it('cannot be used to submit forms', () => {
    const pilter = doShallow({ type: 'submit' })

    expect(pilter).toHaveProp('type', 'button')
  })

  it('passes additional attributes to the element', () => {
    const pilter = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(pilter).toHaveProp('id', 'the-id')
    expect(pilter).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const pilter = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(pilter).not.toHaveProp('className', 'my-custom-class')
    expect(pilter).not.toHaveProp('style')
  })
})
