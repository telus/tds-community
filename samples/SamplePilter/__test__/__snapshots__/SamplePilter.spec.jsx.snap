// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SamplePilter renders 1`] = `
<button
  aria-label="undefined Some Content"
  class="base"
  type="button"
>
  <div
    class="TDS_Box-modules__horizontalPadding-3___2uoUp TDS_Box-modules__verticalPadding-2___1Uh2T"
  >
    <span
      class="TDS_Typography-modules__base___3JQ2A TDS_Typography-modules__wordBreak___11Xyp TDS_Typography-modules__baseFont___-Br-_ TDS_Typography-modules__invertedColor___EXfdx"
    >
      Some Content
    </span>
  </div>
</button>
`;
